diff --git a/dist/utilities/deepCopyObject.js b/dist/utilities/deepCopyObject.js
index 9238a31f41794b441317c78ac5da3dcc22877012..e10431da67906a226c98d3483cbe23b31457bc6c 100644
--- a/dist/utilities/deepCopyObject.js
+++ b/dist/utilities/deepCopyObject.js
@@ -109,7 +109,12 @@ Benchmark: https://github.com/AlessioGr/fastest-deep-clone-json/blob/main/test/b
             if (filterUndefined && v === undefined) {
                 continue;
             }
-            ret[k] = typeof v !== 'object' || v === null ? v : deepCopyObjectSimple(v, filterUndefined);
+            Object.defineProperty(ret, k, {
+                value: typeof v !== 'object' || v === null ? v : deepCopyObjectSimple(v, filterUndefined),
+                enumerable: true,
+                configurable: true,
+                writable: true
+            });
         }
         return ret;
     }
